### ISSUE TITLE
Understanding difference between V1/InflightBatching/InflightFusedBatching

### ISSUE THREAD

**Issue by akhoroshev** [NVIDIA/TensorRT-LLM#478]
We have 3 values
```c++
enum class TrtGptModelType
{
    V1,
    InflightBatching,
    InflightFusedBatching
};
```

What is the V1?
What is the difference between InflightBatching and InflightFusedBatching?

[The support for inflight batching requires the input tensors to be packed (no padding).](https://nvidia.github.io/TensorRT-LLM/gpt_attention.html#inflight-batching). It's ok.

But what about paged kv? I found the [code](https://github.com/NVIDIA/TensorRT-LLM/blob/a21e2f85178111fed9812bb88c2cc7411b25f0ba/cpp/include/tensorrt_llm/runtime/gptModelConfig.h#L154)
```c++
[[nodiscard]] bool constexpr supportsInflightBatching() const noexcept
{
        return mUseGptAttentionPlugin && mInputPacked && mPagedKvCache;
}
```
and it looks like that for InflightBatching paged kv cache needs to be enabled?

**Comment by schetlur-nv** [comment]
Thanks for the question, @akhoroshev ! 
1. V1 batching refers to the traditional batching scheme (i.e. pre in-flight batching) where a batch of requests is taken all the way to completion without including new requests or early exiting completed sequences of the batch in progress . This is simpler to implement since all requests run in lockstep, but commonly leads to wasted computation due to padding.
2. InFlightBatching is a more advanced batching scheme that includes newly arrived requests as soon as the HW has capacity to run it and returns completed requests as soon as the last token is generated, without any need for padding.
3. InFlightFusedBatching is a superset of InFlightBatching and is expected to be better in all cases. It allows for some additional fusion of operations across  requests.

Yes, we require the use of paged KV cache with InflightBatching. Without paged attention, memory pressure is too high and this limits the advantage of advanced batching schemes.

**Comment by lly-zero-one** [comment]
One more question here. From the document, https://nvidia.github.io/TensorRT-LLM/gpt_runtime.html#in-flight-batching-support
`In this release, in-flight batching is supported using separate decoders per request. The biggest difference compared to using a single decoder is in how the token generation from logits is managed. A batch is split into batchSize individual requests and kernels are issued using separated CUDA streams. This behavior may be revisited in a future release to maintain the structure of the batch and improve efficiency.`

Can I assume this description is obsolete? My profiling of in-flight batching shows the attention kernels are always executed on the same cuda stream. Thanks.

**Comment by schetlur-nv** [comment]
'Decoder' here refers to the set of steps that transform output logits to the tokens finally returned to the user (as opposed to the decoder-only transformer architecture). These kernels are still launched in different streams, one for each sequence in the batch.

**Comment by schetlur-nv** [comment]
I am closing this issue. Please feel free to leave a comment here or open a new issue with more questions.
